controllers/flowcollector_controller.go

```diff
@@ -74,9 +78,30 @@ func Start(ctx context.Context, mgr *manager.Manager) error {
 	}
 	r.watcher = watchers.NewWatcher(ctrl)
 
+	// force reconcile at startup
+	go r.InitReconcile(ctx)
+
 	return nil
 }
 
+func (r *FlowCollectorReconciler) InitReconcile(ctx context.Context) error {
+	log := log.FromContext(ctx)
+	log.Info("Initializing resources...")
+
+	var err error
+	for attempt := range initReconcileAttempts {
+		// delay the reconcile calls to let some time to the cache to load
+		time.Sleep(5 * time.Second)
+		_, err = r.Reconcile(ctx, reconcile.Request{})
+		if err != nil {
+			log.Error(err, "Error while doing initial reconcile", "attempt", attempt)
+		} else {
+			break
+		}
+	}
+	return err
+}
```

Comment:
☝️ I wonder if there is an out of box mechanism to trigger the loop after the cache loaded. That's why I'm using a sleep here and this will may work in all situations.

https://redhat-internal.slack.com/archives/C02939DP5L5/p1743518264445429

Reply:
If I remember correctly, when a reconcile loop is failing, you can also return a time value to reschedule the reconciliation.

Reply:
Yeah I gave a try with that without success. 

I'm refactoring the code again to move the static content to another controller wich will be cleaner I guess. I will give another try with the reschedule time on the new controller :+1: